components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/dao/impl/SAML2ArtifactInfoDAOImpl.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.sso.saml.dao.impl;
+
+import org.joda.time.DateTime;
+import org.opensaml.saml2.core.Assertion;
+import org.wso2.carbon.consent.mgt.core.util.JdbcUtils;
+import org.wso2.carbon.database.utils.jdbc.JdbcTemplate;
+import org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException;
+import org.wso2.carbon.identity.base.IdentityException;
+import org.wso2.carbon.identity.core.persistence.JDBCPersistenceManager;
+import org.wso2.carbon.identity.sso.saml.dao.SAML2ArtifactInfoDAO;
+import org.wso2.carbon.identity.sso.saml.dto.SAML2ArtifactInfo;
+import org.wso2.carbon.identity.sso.saml.dto.SAMLSSOAuthnReqDTO;
+import org.wso2.carbon.identity.sso.saml.exception.ArtifactBindingException;
+import org.wso2.carbon.identity.sso.saml.util.SAMLSSOUtil;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInput;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+
+/**
+ * Default implementation of SAML2ArtifactInfoDAO.
+ */
+public class SAML2ArtifactInfoDAOImpl implements SAML2ArtifactInfoDAO {
+
+    @Override
+    public void storeArtifactInfo(SAML2ArtifactInfo saml2ArtifactInfo) throws ArtifactBindingException {
+
+        final String ARTIFACT_INFO_STORE_SQL = "INSERT INTO IDN_SAML2_ARTIFACT_STORE(SOURCE_ID, " +
+                "MESSAGE_HANDLER, AUTHN_REQ_DTO, SESSION_ID, INIT_TIMESTAMP, EXP_TIMESTAMP, ASSERTION_ID) " +
+                "VALUES (?, ?, ?, ?, ?, ?, ?)";
+        JdbcTemplate jdbcTemplate = JdbcUtils.getNewTemplate();
+
+        try {
+            jdbcTemplate.executeInsert(ARTIFACT_INFO_STORE_SQL, (preparedStatement -> {
+                preparedStatement.setBytes(1, saml2ArtifactInfo.getSourceId());
+                preparedStatement.setBytes(2, saml2ArtifactInfo.getMessageHandler());
+                try {
+                    setBlobObject(preparedStatement, saml2ArtifactInfo.getAuthnReqDTO(), 3);
+                } catch (IOException e) {
+                    throw new SQLException(e);
```

Comment:
Can we add a String for better error reporting.
throw new SQLException("Could not set Saml2ArtifactInfo.AuthnReqDTO as a Blob", e);